i18n/cleanUnusedLocaleKeys.js

```diff
@@ -0,0 +1,136 @@
+            // Base case: remove the last key from the current object and return a new object without that key
+            const keyToRemove = keyParts[0]
+            // Use object destructuring with computed property name to extract the property named by 'keyToRemove'
+            // And ignore it (_), while collecting the rest of the properties into a new object called 'rest'.
+            // This effectively creates a new object 'rest' without the property 'keyToRemove'.
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const { [keyToRemove]: _unused, ...rest } = currentObj
+            return rest
+        }
+        const [firstKey, ...remainingKeys] = keyParts
+        // If the first key exists and is an object, recursively remove the nested key
+        if (currentObj[firstKey] && typeof currentObj[firstKey] === 'object' && currentObj[firstKey] !== null) {
+            return {
+                // Return a new object copying all properties from `currentObj`,
+                // But replace the property at `firstKey` with the result of
+                // Recursively removing the key from the nested object at that path.
+                ...currentObj,
+                [firstKey]: removeKeyAtPath(currentObj[firstKey], remainingKeys)
+            }
+        }
+
+        // If the key path does not exist or is not an object, return the object unchanged
+        return currentObj
+    }
+
+    // For each key path to remove, split it into parts and call the recursive function to remove it
+    return keysToRemove.reduce((updatedObj, keyPath) => {
+        const keyParts = keyPath.split('.')
+        return removeKeyAtPath(updatedObj, keyParts)
+    }, obj)
+}
+
+const run = async () => {
+    const usedKeys = await extractUsedTranslationKeys()
+
+    const unusedKeysByLocale = translationFilePaths.reduce((acc, filePath) => {
+        const content = JSON.parse(fs.readFileSync(filePath, 'utf-8'))
+        const allKeys = extractAllKeysFromJson(content)
+        const unusedKeys = allKeys.filter(key => !usedKeys.has(key))
+
+        if (unusedKeys.length > 0) {
+            // Clean JSON
+            const cleanedContent = removeKeysFromJson(content, unusedKeys)
+            // Rewrite file
+            fs.writeFileSync(filePath, JSON.stringify(cleanedContent, null, 2), 'utf-8')
+            acc[path.basename(filePath)] = unusedKeys
+        }
+        return acc
+    }, {})
+
+    if (Object.keys(unusedKeysByLocale).length === 0) {
+        console.info('✅ No unused translation keys found.')
+        return
+    }
+
+    const output = Object.entries(unusedKeysByLocale)
+        .map(([file, keys]) => {
+            console.info(`📂 ${file}:`)
+            keys.forEach(key => console.info(`  - ${key}`))
+            return `${file}:\n${keys.map(k => `- ${k}`).join('\n')}\n`
+        })
+        .join('\n')
+
+    fs.writeFileSync('potentially_unused_keys.txt', output, 'utf-8')
```

Comment:
let's clarify these are i18n keys, since there are other keys in the project

```suggestion
    fs.writeFileSync('potentially_unused_i18n_keys.txt', output, 'utf-8')
```